core/trino-main/src/main/java/io/trino/operator/scalar/JoniRegexpFunctions.java

```diff
@@ -408,4 +409,14 @@ public static long regexpCount(@SqlType("varchar(x)") Slice source, @SqlType(Jon
 
         return count;
     }
+
+    public static int getSearchingOffset(Matcher matcher, int at, int range)
+    {
+        try {
+            return matcher.searchInterruptible(at, range, Option.DEFAULT);
+        }
+        catch (InterruptedException interruptedException) {
+            throw new TrinoException(GENERIC_USER_ERROR, "Regular expression matching was interrupted, likely because it took too long. Regular expression in the worst case can have a catastrophic amount of backtracking and having exponential time complexity");
```

Comment:
```suggestion
            Thread.currentThread().interrupt();
            throw new TrinoException(GENERIC_USER_ERROR, "Regular expression matching was interrupted, likely because it took too long. Regular expression in the worst case can have a catastrophic amount of backtracking and having exponential time complexity");
```